nucypher/policy/conditions/evm.py

```diff
@@ -19,13 +23,27 @@
 from nucypher.policy.conditions.lingo import ReturnValueTest
 from nucypher.policy.conditions.utils import CamelCaseSchema, camel_case_to_snake
 
+# TODO: Move this to a more appropriate location,
+#  but be sure to change the mocks in tests too.
 # Permitted blockchains for condition evaluation
-_CONDITION_CHAINS = (
-    1,     # ethereum/mainnet
-    5,     # ethereum/goerli
-    137,   # polygon/mainnet
-    80001  # polygon/mumbai
-)
+from nucypher.utilities import logging
+
+_CONDITION_CHAINS = {
+    1: "ethereum/mainnet",
+    5: "ethereum/goerli",
+    11155111: "ethereum/sepolia",
+    137: "polygon/mainnet",
+    80001: "polygon/mumbai",
+    10: "optimism/mainnet",
+    69: "optimism/kovan",
+    420: "optimism/goerli",
+    42161: "arbitrum/mainnet",
+    421613: "arbitrum/goerli",
+    100: "gnosis/mainnet",
+    10200: "gnosis/chiado",
+    534352: "scroll/mainnet",
+    534351: "scroll/sepolia",
+}
```

Comment:
While I totally agree with the direction of the work on this PR, I want to be cautious on this specific set of lines. Let's validate them before actually adding them to the allowlist. We need criteria to evaluate both:
(1) Demand-side: To have evidence adopters requesting some chain are "sponge-worthy" (i.e. not because someone claiming to use TACO needs some chain, we're going to add it). 
(2) Supply-side: make sure that stakers/node operators can handle the overhead of adding more chains to the list. In some cases this can be supported by "free plans" (e.g., from this list, Infura supports free add-ons for polygon, optimism and arbitrum, although threre's the lingering question about how much can we lean on Infura), but what about Gnosis and Scroll, for instance?

Reply:
Original issue tracking this:
https://github.com/nucypher/tdec/issues/7

Reply:
Since these are for reads, I very much prefer for the default path not to be Infura / another centralized solution.  We can provide RPC endpoints as defaults.  If somebody wants to use their Infura, we can make that easy too.

But this is a _very_ easy place to provide a default that's more decentralized.

As for the specifics: Gnosis and Scroll seems like obvious defaults.  Gnosis is the minting ground for POAP, which I think we want to support out of the gate.  Scroll is a ZK insurgent which Electora used successfully in their hackathon win.

And although the loss of the Today Sponge is a tragedy, I don't think we're at risk of running out of multichain RPC endpoints.

It's more like "Top of the Multichain to you!"

(with the exclamation point)

Reply:
I'm suggesting gnosis and scroll be allowed here since there are adopters actively using both, fortunately there are several officially maintained RPC endpoints for both networks.  An infura integration can ease some concerns here, but I'd also like to suggest we include public endpoint lists maintained by third parties like chainlist.network.

if there's more validation to do before adding to this list we can remove it, however I'd like to deploy a gnosis-supporting testnet asap for use with the revealer adopter asap.

Reply:
A bigger issue this illuminates is if we'll require all nodes to support all endpoints or not.  If nodes can select only a subset, we will need to build out an endpoint for use during sampling to query which nides support which chains... Which itself raises questions about economic schemes, etc.

Reply:
> A bigger issue this illuminates is if we'll require all nodes to support all endpoints or not. If nodes can select only a subset, we will need to build out an endpoint for use during sampling to query which nides support which chains... Which itself raises questions about economic schemes, etc.

No, I don't think we want pick-and-choose endpoints for v7. As for the future, we we would have to solve first the questions you mention. But it's definitely something worth exploring and I think we already mention this in some issue (I'll try to find it later).

I suggest that for the moment let's focus here on those chains that will be 100% required for all nodes, and that should satisfy both demand- and supply-side requirements. We know it is the case for Ethereum and Polygon. Is this the case for Gnosis chain and Scroll? Just to be clear, I'm not asking this rhetorically: I genuinely would like to see what's the evidence supporting both requirements.

> @jmyles: [...] But this is a very easy place to provide a default that's more decentralized.
> @KPrasch: [...] fortunately there are several officially maintained RPC endpoints for both networks.

These two comments are IMO contradictory.

Reply:
Yeah, I'm not sure how to think about "optional" chains.  Also, since nodes only stake on Ethereum mainnet, it's not immediately obvious how we'd slash for bad behavior in the future on other chains (though of course there are ways, just not immediately obvious ones).

That said, this list seems like a good list of defaults.

I think @KPrasch was thinking that my record release was using conditions on Gnosis when he wrote the above, but that's not so - only Ethereum mainnet (though it depends on merge of this logic just the same!).  

However, I can think of an imminent use for Gnosis: using POAPs at events where we're presenting for the purposes of live demos.  In fact, I can imagine doing just this at the Encode hackathon in a couple days.

Reply:
✔️